java/java/src/main/java/org/whispersystems/libsignal/state/impl/InMemorySessionStore.java

```diff
@@ -34,6 +35,23 @@ public synchronized SessionRecord loadSession(SignalProtocolAddress remoteAddres
     }
   }
 
+  @Override
+  public synchronized List<SessionRecord> loadExistingSessions(List<SignalProtocolAddress> addresses) throws NoSessionException {
+    List<SessionRecord> resultSessions = new LinkedList<>();
+    for (SignalProtocolAddress remoteAddress : addresses) {
+      byte[] serialized = sessions.get(remoteAddress);
+      if (serialized == null) {
+        throw new NoSessionException("no session for " + remoteAddress);
```

Comment:
It would be considered a bad state if one of the remote clients was missing a session, and would prevent loading any of the sessions? It is all-or-nothing?

Reply:
Yeah, this API is meant for the fast path of "I already have sessions with all these devices", something we can be assured of because the payload is either a SignalMessage or PreKeySignalMessage (which came from a 1:1 session), or a SenderKeyMessage (which must have previously had its key sent in a 1:1 distribution message). That is, if apps are hitting this exception, there's a bug.